app/identifiers/register/trustees/partnership/PartnershipAddressYearsId.scala

```diff
@@ -43,20 +46,25 @@ case class PartnershipAddressYearsId(index: Int) extends TypedIdentifier[Address
 object PartnershipAddressYearsId {
   override lazy val toString: String = "partnershipAddressYears"
 
-  implicit val cya: CheckYourAnswers[PartnershipAddressYearsId] = {
-
+  implicit def cya(implicit countryOptions: CountryOptions, messages: Messages, ua: UserAnswers): CheckYourAnswers[PartnershipAddressYearsId] =
     new CheckYourAnswers[PartnershipAddressYearsId] {
-      val label: String = "messages__checkYourAnswers__trustees__partnership__address_years"
-      val changeAddressYears: String = "messages__visuallyhidden__trustee__address_years"
+      override def row(id: PartnershipAddressYearsId)(changeUrl: String, ua: UserAnswers): Seq[AnswerRow] = {
+        def trusteeName(index: Int) = ua.get(PartnershipDetailsId(index)).fold(messages("messages__theTrustee"))(_.name)
+
+        def label(index: Int) = messages("messages__hasBeen1Year", trusteeName(index))
```

Comment:
label and changeAddressYears can be a val instead of def as index is already available inside row

Reply:
Fixed